libc/functions/i/ioctl.part-impl.md

```diff
@@ -0,0 +1,81 @@
+# Synopsis 
+`#include <ioctl.h>`</br>
+
+` int ioctl(int fildes, int request, ... /* arg */);`</br>
+
+## Status
+Partially implemented
+## Conformance
+IEEE Std 1003.1-2017
+## Description
+
+
+The `ioctl()` function shall perform a variety of control functions on devices. For _fildes_ not referring to devices, the functions performed by this call are unspecified.
+
+The _fildes_ argument is an open file descriptor that refers to a device.
+
+The _request_ argument selects the control function to be performed and shall depend on the device being addressed.
+
+The _arg_ argument represents additional information that is needed by this specific device to perform the requested function. The type of _arg_ depends upon the particular control request, but it shall be either an integer or a pointer to a device-specific data structure.
+
+In order to define driver request one should use one of the macros presented below, which are available in the `ioctl.h` header file:
+
+    _IOC(inout,group,num,len) - macro which return every possible request
+
+_inout_ - use these defines: IOC_VOID (no data), IOC_OUT (read), IOC_IN (write), IOC_INOUT (read/write)
+_group_ - describe group type (collecting the ioctls in groups for a common purpose or a common driver), 8-bit value
+_num_ - specifie serial number of request, 8 -bit value
+_len_ - how big is size of data, 14-bit value
+
+    _IOV(g,n) - macro which return no data transmission request
+    _IOR(g,n,t) - macro which return read request
+    _IOW(g,n,t) - macro which return write request
+    _IOWR(g,n,t) - macro which return read/write request
+
+_g_ - 8-bit group value
+_n_ - 8-bit serial number
+_t_ - type of data
+
+There are also some checking macros:
+
+    OCPARM_LEN(x) - returns data length
+    IOCBASECMD(x) - returns base command value (2 least significant bytes describing group and serial number)
+    IOCGROUP(x) - returns group type value
+
+The resulting requst will be in such form: 2 bits describe direction (00: none, 10: write, 01: read, 11: read/write) followed by 14 size bits, followed by an 8-bit group type (collecting the ioctls in groups for a common purpose or a common driver) and an 8-bit serial number.
+
+It is worth to note that to get data send to driver inside message, one should use `ioctl_unapck()`.
+
+
+## Return value
+
+
+Upon successful completion, `ioctl()` shall return a value other than -1 that depends on the device control function. Otherwise, it shall return -1 and set `errno` to indicate the error.
```

Comment:
```suggestion
Upon successful completion, `ioctl()` shall return a value other than -1 that depends on the device control function. Otherwise, it shall return `-1` and set `errno` to indicate the error.
```